xdg-terminal-exec

```diff
@@ -290,11 +278,86 @@ read_entry_path() {
+DATA_PREFIX_DIR=xdg-terminals
+# Name of config files to search and read
+CONFIG_NAME=xdg-terminals.list
+
+# Directory hierarchy searched for terminal config and desktop entry files
+# TODO: Consider using $XDG_DATA_DIRS
+DATA_HIERARCHY="${XDG_DATA_HOME:-$HOME/.local/share}:${XDG_DATA_DIRS:-/usr/local/share:/usr/share}"
+CONF_HIERARCHY="${XDG_CONFIG_HOME:-$HOME/.config}:${XDG_CONFIG_DIRS:-/etc/xdg:/usr/etc/xdg}"
+
+# All desktop entry paths in descending order of preference
+ENTRY_PATHS=''
+# All desktop entry ids in descending order of preference
+ENTRY_IDS=''
+
+# Generate list of possible config files for current DE
+# TODO: Should this be case insensitive? Desktop entry checks that also use $XDG_CURRENT_DESKTOP are not
+lowercase_desktops="$(printf '%s' "${XDG_CURRENT_DESKTOP-}" | tr '[:upper:]' '[:lower:]')"
+debug "lowercase desktops are: '$lowercase_desktops'"
+# All config files are searched for, and read immediatelly, rather than when out of IDs
+# This way all possible IDs are know before iterating over them, and are in order of preference
+# Further down, duplicate IDs are then removed using `awk`
+# This avoids having to use cumbersome blacklists, or fragile string operations, to deal with them
+IFS=':'
+for config_dir in $CONF_HIERARCHY; do
+	# Append read ID(s) to list, using brace expansion for better readability
+	IFS=':'
+	for desktop in $lowercase_desktops; do
+		IFS="$OIFS"
+		ENTRY_IDS="${ENTRY_IDS}${N}$(read_config_path "$config_dir"/"$desktop"-"$CONFIG_NAME")"
+	done
+	IFS="$OIFS"
+	ENTRY_IDS="${ENTRY_IDS}${N}$(read_config_path "$config_dir"/"$CONFIG_NAME")"
+done
+
+# Generate list of possible terminal entry files and also add their IDs to
+IFS=':'
+for data_dir in $DATA_HIERARCHY; do
+	IFS="$OIFS"
+	if [ -d "$data_dir"/"$DATA_PREFIX_DIR" ]; then
+		debug "searching in '$data_dir/$DATA_PREFIX_DIR'"
+		while read -r entry_path; do
+			debug "found desktop entry '$entry_path'"
+			# Remove data directory path
+			entry_path="${entry_path#"$data_dir"/"$DATA_PREFIX_DIR"/}"
+			# Create entry ID from path
+			entry_id="$(printf '%s' "$entry_path" | tr '/' '-')"
+			debug "adding matching id '$entry_id'"
+			# Append found ID to list, using brace expansion for better readability
+			ENTRY_IDS="${ENTRY_IDS}${N}${entry_id}"
+			# Prepend data directory path, and delimit its end using an extra '/' character
+			# This is done so the ID can be recreated in `find_entry_path`
+			# Other possible way of storing the data direcotry could be:
+			# $data_dir/$DATA_PREFIX_DIR/$entry_path/$entry_id
+			# Where '.desktop/' is the delimiter, as IDs do not contain '/', and entry files must end in '.desktop'
+			entry_path="$data_dir"/"$DATA_PREFIX_DIR"//"$entry_path"
+			# Append found path to list, using brace expansion for better readability
+			ENTRY_PATHS="${ENTRY_PATHS}${N}${entry_path}"
+			# TODO: Print order of found files is unpredictable, sort them?
```

Comment:
yes, sort it

Reply:
Done
For future reference, `| xargs -0r printf '%s\n'` could be used, if `-n 1` is found to be undesirable
Also the change should be noted in the README. but I'll leave that to another PR